### PR TITLE
feat: Votes page horizontal scrollable on mobile

### PR DESCRIPTION
**Pull request by ChaituVR**
### Summary

Closes: #331 

- Adds a horizontal scroll for votes page on mobile
- Hide right side bar (results and choices) on mobile (sticky needs a fixed height)
- Remove padding at bottom for votes page (sticky needs a fixed height)
- On swipe, it should not open right sidebar https://github.com/snapshot-labs/sx-monorepo/pull/618#discussion_r1716404917

### How to test

1. Go to votes page in any proposal https://bafybeihxvycavj4u7au7kmcwgwyp5islxrzj562hoskasnyieqgkncpbma.on.fleek.co/#/s:safe.eth/proposal/0x8d97075eabfdf3fb9f43f3b490e8e5a7326d77bee919b78543621e36e5b2d233/votes
2. Everything should work same as before on desktop
3. On mobile, you should see a horizontal scroll

### CODE REVIEW

File: apps/ui/src/views/Proposal/Votes.vue
Comment (bonustrack) [inline apps/ui/src/views/Proposal/Votes.vue:35]: Can't you make it just with CSS?

Comment (ChaituVR) [reply apps/ui/src/views/Proposal/Votes.vue]: @bonustrack agreed that we can move on, it will be a bit slow if we run yarn dev but should be fine on prod
If someone is interested in solving I have created this https://jsbin.com/yihacif/edit?html,css,output

**Comment by wa0x6e** [comment]
Screenshot 2024-08-15 at 00 50 26

On desktop, I see x and y scrollbar by default for the votes table

the vertical scrollbar is just moving by 1px

**Comment by wa0x6e** [comment]
On desktop, I now see 3 scrollbar:

Screenshot 2024-08-15 at 00 53 45

The whole page scrollbar is just moving by one pixel

And there's now an x and y scrollbar for the votes table

**Comment by wa0x6e** [comment]
On < 1000px screen, there's even 2 horizontal scrollbar

Screenshot 2024-08-15 at 01 00 06

**Comment by ChaituVR** [comment]
Scrollbars inside votes table effecting the widths, using no-scrollbar to hide them and fixed the issue with page level scroll bar

File: apps/ui/src/views/Proposal/Votes.vue
Comment (wa0x6e) [inline apps/ui/src/views/Proposal/Votes.vue:125]: Is this div and the one above can not be merged together ?

Comment (ChaituVR) [reply apps/ui/src/views/Proposal/Votes.vue:125]: The one in Proposal.vue?

Comment (ChaituVR) [reply apps/ui/src/views/Proposal/Votes.vue:125]: The top one is allowing us to overflow and the child is setting fixed width to be scrolled horizontally, and settings fixed height for `sticky` to work, and a `overflow-x-auto` is preventing this div going off when we scroll down

so merging seems not possible with the conditions we have 🙈

Comment (ChaituVR) [reply apps/ui/src/views/Proposal/Votes.vue:129]: There is a solution, added a fix

**Comment by wa0x6e** [comment]
When trying to horizontal scroll when the element has reached the end, it spilled over adjacent div

https://www.loom.com/share/cc19824d20214233ad334cafcd08b07d?sid=4f86dd92-0c11-44d1-a47d-17c153f024ca

Maybe add a overflow hidden somewhere to prevent that

**Comment by wa0x6e** [comment]
> When trying to horizontal scroll when the element has reached the end, it spilled over adjacent div
> 
> https://www.loom.com/share/cc19824d20214233ad334cafcd08b07d?sid=4f86dd92-0c11-44d1-a47d-17c153f024ca
> 
> Maybe add a overflow hidden somewhere to prevent that

the votes table is spilling over the left and right sidebar

File: apps/ui/src/views/Proposal/Votes.vue
Comment (wa0x6e) [inline apps/ui/src/views/Proposal/Votes.vue:380]: We can avoid this style by using the class `h-[calc(100vh-113px)]`

Comment (ChaituVR) [reply apps/ui/src/views/Proposal/Votes.vue:380]: tq tq

**Comment by bonustrack** [comment]
We should have a native window scroll and not a div scroll
image

**Comment by ChaituVR** [comment]
> We should have a native vertical window scroll and not a div scroll image

without second sticky header we can do all this . for sticky to work (With `overflow-x-auto`) we need a fixed height div 😢

**Comment by bonustrack** [comment]
Coinmarketcap seem to have solved that, they are using table, is there something you can take from their approach?

**Comment by ChaituVR** [comment]
> When trying to horizontal scroll when the element has reached the end, it spilled over adjacent div
> 
> https://www.loom.com/share/cc19824d20214233ad334cafcd08b07d?sid=4f86dd92-0c11-44d1-a47d-17c153f024ca
> 
> Maybe add a overflow hidden somewhere to prevent that

This happens before as well, it is because of the sticky property, also happens in many parts of app, we need to use `bg-skin-bg` adjust z-index of all sidebars and nav bars

example:
Untitled 2

File: apps/ui/src/components/Ui/ScrollerHorizontal.vue
Comment (Sekhmet) [inline apps/ui/src/components/Ui/ScrollerHorizontal.vue:2]:
```suggestion
const emit = defineEmits<{
  (e: 'scroll', target: HTMLElement);
}>();
```

**Review by Sekhmet — COMMENTED** [review]
Overall it looks good to me, but as always I'm not huge fan of hiding scrollbars for looks sacrificing UX (on certain window sizes content becomes scrollable and it's not obvious because we don't show scrollbars) and you have to know that you can scroll it and do so by `Shift + scroll`:
image

**Comment by wa0x6e** [comment]
> Overall it looks good to me, but as always I'm not huge fan of hiding scrollbars for looks sacrificing UX (on certain window sizes content becomes scrollable and it's not obvious because we don't show scrollbars) and you have to know that you can scroll it and do so by `Shift + scroll`: image

Agree, on some screen size, columns match perfectly to the cut, so we have no way of knowing there’s more element hidden, and we have to scroll.

we could add the gradient to highlight the fact it’s scrollable

**Comment by ChaituVR** [comment]
> Overall it looks good to me, but as always I'm not huge fan of hiding scrollbars for looks sacrificing UX (on certain window sizes content becomes scrollable and it's not obvious because we don't show scrollbars) and you have to know that you can scroll it and do so by Shift + scroll

Agree but we won't be able to show scrollbar here (div doesn't have a fixed height), any other ideas to improve ux?

**Comment by Sekhmet** [comment]
@bonustrack this PR can be merged after your approval.

**Review by bonustrack — APPROVED** [review]
tACK

**Pull request merged**
